src/extractor/edge_based_graph_factory.cpp

```diff
@@ -431,6 +431,23 @@ void EdgeBasedGraphFactory::GenerateEdgeExpandedEdges(
                                                 sizeof(target_node.weight));
                         previous = target_node.node_id;
                     }
+
+                    // We also now write out the mapping between the edge-expanded edges and the
+                    // original nodes. Since each edge represents a possible maneuver, external
+                    // programs can use this to quickly perform updates to edge weights in order
+                    // to penalize certain turns.
+
+                    const auto &from_node = m_node_info_list[node_u];
+                    const auto &via_node =
+                        m_node_info_list[m_node_based_graph->GetTarget(edge_from_u)];
+                    const auto &to_node = m_node_info_list[m_node_based_graph->GetTarget(turn.eid)];
```

Comment:
I don't think `to_node` is correct in all cases here.

The `node_based_graph->GetTarget` gets the *last* node in a node-based-edge.  In the case where this is a simple edge, with just one segment, we'll get the correct value here.

However, in the case where the node-based-edge is compressed, what we want is the *second* node.

To get `to_node`, we'll need to do a lookup in the compressed geometry container, get the compressed geometry bucket, then get the first `node_id` in that list.  This will work for all types of edges.

You can write a test case for this by creating ways that have more than two segments, like:

```
| a | b | c | d |
| e |   |   |   |
| f |   |   |   |

Ways:
abcd
aef
```